modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/ArtifactAwareContextSelector.java

```diff
@@ -94,23 +100,43 @@ public void removeContext(LoggerContext context) {
    * @return the {@link ClassLoader} owner of the logger context
    */
   static ClassLoader resolveLoggerContextClassLoader(ClassLoader classLoader) {
-    ClassLoader loggerClassLoader = classLoader == null ? Thread.currentThread().getContextClassLoader() : classLoader;
+    WeakReference<ClassLoader> loggerClassLoaderReference = classLoaderMap.get(classLoader);
+    ClassLoader loggerClassLoader;
+
+    if (loggerClassLoaderReference == null) {
+      loggerClassLoader = getLoggerClassLoader(classLoader);
```

Comment:
even though the map is synchronized, this code may be executed concurrently and the loggerClassLoader calculated extra times needlessly.

Reply:
sure, but that is a minor performance overhead, I can't think of any solution that will avoid that duplicated calculation without requiring a locking, which will be worst that this solution. Do you have an alternative solution?

Reply:
disregard my previous comment...